libs/cardano-ledger-core/src/Cardano/Ledger/Plutus/ToPlutusData.hs

```diff
@@ -77,8 +78,10 @@ instance ToPlutusData NonNegativeInterval where
   fromPlutusData _ = Nothing
 
 instance ToPlutusData CostModels where
-  toPlutusData cmdls = toPlutusData (flattenCostModels cmdls)
-  fromPlutusData x = mkCostModelsLenient <$> fromPlutusData x
+  toPlutusData costModels = toPlutusData $ fmap toInteger <$> flattenCostModels costModels
+  fromPlutusData costModelsData = do
+    costModels :: Map.Map Word8 [Integer] <- fromPlutusData costModelsData
+    mkCostModelsLenient (Map.map (map fromInteger) costModels)
```

Comment:
Is the `fromInteger` safe wrt overflows?

Reply:
Yes, because `fromPlutusData` is only used for testing. There will never be a case when we need to translate PlutusContext back to the ledger representation, other than for testing the forward translation